### PR TITLE
sinkv2(ticdc): make event sink close waiting for all callbacks be called

### PR DESCRIPTION
**Pull request by hi-rustin**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tiflow/issues/5928

### What is changed and how it works?

- Make event sink close waiting for all callbacks be called. Otherwise we would have a double-writing problem.
- sinkv2(ticdc): remove useless replicaID

#### Tests 

 - Unit test

##### Will it cause performance regression or break compatibility?
No
##### Do you need to update user documentation, design documentation or monitoring documentation?
No
### Release note 

```release-note
None
```

### CODE REVIEW

File: cdc/sinkv2/tablesink/event_table_sink.go [hunk 1]
```diff
@@ -103,6 +104,10 @@ func (e *eventTableSink[E]) GetCheckpointTs() model.ResolvedTs {
 
 func (e *eventTableSink[E]) Close() {
 	e.state.Store(pipeline.TableStateStopped)
+	for e.progressTracker.trackingCount() != 0 {
```
Comment (hicqu) [inline cdc/sinkv2/tablesink/event_table_sink.go:107]: Is it necessary? I think just ensure no more events will be flushed into downstreams after `Close` is called is enough.

File: cdc/sinkv2/tablesink/event_table_sink.go [hunk 2]
```diff
@@ -103,6 +104,10 @@ func (e *eventTableSink[E]) GetCheckpointTs() model.ResolvedTs {
...
+		// Sleep for a while to prevent CPU spin.
+		time.Sleep(time.Second)
```
Comment (hicqu) [inline cdc/sinkv2/tablesink/event_table_sink.go:109]: Maybe 1s is too long. How about 100ms?

File: cdc/sinkv2/tablesink/progress_tracker.go [hunk 3]
```diff
@@ -118,3 +118,11 @@ func (r *progressTracker) minTs() model.ResolvedTs {
 
 	return r.lastMinResolvedTs
 }
+
+// trackingCount returns the number of pending events and resolved tss.
+// Notice: must hold the lock.
+func (r *progressTracker) trackingCount() int {
```
Comment (hicqu) [inline cdc/sinkv2/tablesink/progress_tracker.go:124]: We can add a `closed` flag into `progressTracker` to simplify `remove` and make `minTs()` more safe.

**Review by hicqu — APPROVED** [review]

**Review by asddongmen — APPROVED** [review]

**Pull request merged**